### PR TITLE
Re-ordering conditions once they are deleted

### PR DESCRIPTION
**Pull request by mmumby**
- Updated BE to re-order conditions after they are deleted.

### CODE REVIEW

File: services/core-api/app/api/mines/permits/permit_conditions/resources/permit_conditions_resource.py [hunk 1]
```diff
@@ -96,4 +96,22 @@ def delete(self, mine_guid, permit_guid, permit_amendment_guid, permit_condition
 
         permit_condition.save()
 
+        if permit_condition.parent_permit_condition_id is not None:
+            parent = PermitConditions.find_by_permit_condition_id(
+                permit_condition.parent_permit_condition_id)
```
Comment (Jsyro) [inline services/core-api/app/api/mines/permits/permit_conditions/resources/permit_conditions_resource.py:101]: Why don't we just traverse 'permit_condition.parent_condition'. Shouldn't need a whole lookup.

File: services/core-api/app/api/mines/permits/permit_conditions/resources/permit_conditions_resource.py [hunk 2]
```diff
@@ -96,4 +96,22 @@ def delete(self, mine_guid, permit_guid, permit_amendment_guid, permit_condition
...
+            sorted_conditions = sorted(parent.sub_conditions, key=lambda x: x.display_order)
+            for i, condition in enumerate(sorted_conditions):
+                condition.display_order = i + 1
+            parent.save()
+        else:
+            conditions = PermitConditions.find_all_by_permit_amendment_id(
+                permit_condition.permit_amendment_id)
```
Comment (Jsyro) [inline services/core-api/app/api/mines/permits/permit_conditions/resources/permit_conditions_resource.py:108]: Might want to load permit_amendment then get conditions member. but i suppose they are equivalent.

File: services/core-api/app/api/mines/permits/permit_conditions/resources/permit_conditions_resource.py (hunk 1 above)
Comment (nayr974) [reply services/core-api/app/api/mines/permits/permit_conditions/resources/permit_conditions_resource.py:101]: .parent didn't seem accessible outside of the scope of the model itself. Maybe a get_parent method on the model might work

Comment (Jsyro) [reply services/core-api/app/api/mines/permits/permit_conditions/resources/permit_conditions_resource.py:101]: PermitCondition should get a relationship to it's parent (just like relationship to it's set of children). Also, that model should be singular.

**Pull request merged**